src/utils/string/StringUtils.js

```diff
@@ -14,7 +14,7 @@ import {getAnonymousUserId} from '../users';
 import {getUserId} from '../token';
 
 const NO_POLL_UUID = '00000000-0000-0000-0000-000000000000';
-
+const urlRegex = /(https?:\/\/\S+)+/g;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `urlRegex` regular expression is not robust enough to validate URLs. It only checks if the string starts with "http://" or "https://", followed by one or more non-space characters. This could lead to false positives as it would consider strings like "http://a" or "https://!" as valid URLs, which they are not. Consider using a more comprehensive regex for URL validation.
```diff
- const urlRegex = /(https?:\/\/\S+)+/g;
+ const urlRegex = /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(\:\d+)?(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->